lib/dal/src/attribute/value.rs

```diff
@@ -2397,30 +2419,66 @@ impl AttributeValue {
         ctx: &DalContext,
         id: AttributeValueId,
     ) -> AttributeValueResult<Vec<AttributeValueId>> {
-        let workspace_snapshot = ctx.workspace_snapshot()?;
+        if let Some(prop) = Self::prop_for_id(ctx, id).await? {
+            match prop.kind {
+                PropKind::Boolean | PropKind::Integer | PropKind::Json | PropKind::String => {
+                    Ok(vec![])
+                }
+                PropKind::Array | PropKind::Map => {
+                    Self::get_child_av_ids_from_ordering_node(ctx, id)
+                        .await?
+                        .ok_or(AttributeValueError::NoOrderingNodeForAttributeValue(id))
+                }
+                // Unlike maps or arrays, we want to walk object
+                // attribute values in prop order, not attribute
+                // value order, so that we always return them in the
+                // same order (the order the props were created for
+                // the schema variant), not the order they were set
+                // on the attribute value
+                //
+                // TODO doing this on read papers over the fact that we're storing them in an
+                // order we do not prefer. We should really just ensure it's impossible to
+                // create a node with these out of sync, or remove the ordering node from
+                // object props altogether.
```

Comment:
do you remove remove the ordering node from object attributes?

Reply:
From object AttributeValues, yeah.